OpenRA.Mods.Common/Widgets/ScrollPanelWidget.cs

```diff
@@ -347,11 +347,11 @@ public void Bind(IObservableCollection c, Func<object, Widget> makeWidget, Func<
 			{
 				if (collection != null)
 				{
-					collection.OnAdd -= BindingAdd;
-					collection.OnRemove -= BindingRemove;
-					collection.OnRemoveAt -= BindingRemoveAt;
-					collection.OnSet -= BindingSet;
-					collection.OnRefresh -= BindingRefresh;
+					collection.OnAdd -= (item) => BindingAdd(item, collection);
```

Comment:
Fixing this should be easy.  Change the `IObservableCollection` implementations to pass the second argument (`this` in that context) instead of DIYing it here.